### PR TITLE
Add global norm prod kernels to dev

### PR DESCRIPTION
**Pull request by gordicaleksa**
Add the faster kernels we use in "prod" to our "dev" kernels.

### CODE REVIEW

File: dev/cuda/common.h [hunk 1]
```diff
@@ -5,6 +5,8 @@
 #include <cublasLt.h>
 #include <float.h>
 
+#define WARP_SIZE 32U
+extern cudaDeviceProp deviceProp;
```
Comment (ngc92) [inline dev/cuda/common.h:9]: this seems like a dangerous change? You are promising the existence of a global variable deviceProp to all our dev kernels, but common.h  does not provide such a thing.
@gordicaleksa

Comment (gordicaleksa) [reply dev/cuda/common.h:9]: We do the same thing in `llmc/cuda_common.h` afaiu - this just makes the variable globally accessible to all kernels in a specific file, but it's that file's responsibility to initiate it if it needs to use it.

I think that's how we do it elsewhere, e.g. in `train_gpt2.cu` we initialize it in the beginning of the main func (declaration is imported from `cuda_common.h`).

Comment (ngc92) [reply dev/cuda/common.h:9]: there  are some important differences between the two situations:
in `llmc/cuda_common.h`, we expect that this header file will be included over multiple paths into the final executable, so we need to take care to ensure that we don't introduce duplicate definitions: hence, every function is marked inline, and variables are marked extern (in principle, we could mark them inline as well, C++17 makes inline available to variables). There is exactly one place where this extern variable gets defined, in `train_gpt2.cu` (that definition gets dragged into  profile and test due to the weird include thing we do). So for every header file in the llmc directory, you can use ´deviceProp` , and are guaranteed to get an actual object there.

In contrast, in dev, the header promises a  deviceProp, but that promise is broken in all but one file, leading to linker errors if you actually tried to use it.

Given that none of the functions are marked inline, I'm not sure if it actually makes sense to declare the device properties as extern. With the current code, maybe the best way would be to define them in that header, and populate them in `setup_main`

Comment (gordicaleksa) [reply dev/cuda/common.h:9]: Ok, makes sense! Do you want to propose a patch? Happy to review & fix it

**Pull request merged**